src/Dialect/ONNX/ONNXOps/OpHelper.cpp

```diff
@@ -539,6 +539,12 @@ bool isDenseONNXConstant(Value result) {
   if (!isa_and_nonnull<ElementsAttr>(constOp.getValueAttr()))
     return false;
 
+  // Except DenseResourceElementsAttr is too hard to work with, since it
+  // sometimes has a different shape and element type than constOp, plus
+  // DenseResourceElementsAttr has a limited API.
+  if (isa<DenseResourceElementsAttr>(constOp.getValueAttr()))
+    return false;
+
```

Comment:
You might need to edit Line 538 too that contains DenseResourceElementsAttr.

Reply:
what I tried to express with the comment is that it's a two step process: 1st in lines 537-540 we require that 'value' is an ElementsAttr (which could be a DenseResourceElementsAttr), 2nd in lines 542-546 we enforce the further requirement that 'value' is not a DenseResourceElementsAttr (because it can behave in strange ways)